compiler/z/codegen/S390Instruction.cpp

```diff
@@ -5804,17 +5804,17 @@ TR::S390VirtualGuardNOPInstruction::generateBinaryEncoding()
    // in) if the patching occurs during GC pause times.  The patching of up to 6-bytes is potentially
    // not atomic.
 
-   bool performEmptyPatch = getNode()->isHCRGuard() || getNode()->isProfiledGuard() || getNode()->isOSRGuard();
+   bool performEmptyPatch = cg()->isStopTheWorldGuard(getNode()) || getNode()->isProfiledGuard();
```

Comment:
Outside the scope of this item but just confirming with the experts here that profiled guards are *not* stop the world guards? In which case I'm not sure this code is safe because we would not emit any instructions for profiled guards.

Reply:
The code is safe but needs cleanup. The concept of stop the world guard is only relevant for nopable guards which are nops got patched at runtime. Profiled guards are not nopable guards and they are explicit compare and jump instructions. Therefore, the getNode()->isProfiledGuard() should never return true when calling from S390VirtualGuardNOPInstruction.

Reply:
Sry, what I said about isProfiledGuard() can never be true is wrong. When a profiled guard is merged with HCRGuard both explicit compare instructions and the S390VirtualGuardNOPInstruction are generated for that profiled guard. The gist for this code is that when this happens the explicit compare instruction can be a patched point and nothing needs to be generated for S390VirtualGuardNOPInstruction. See the comments here: https://github.com/eclipse/omr/blob/94eaf838db94badd7a7c57bebc621994a471cf11/compiler/z/codegen/S390Instruction.cpp#L5809-L5811